search/api.py

```diff
@@ -51,7 +53,39 @@ def construct_queryset(repo_slug, query='', selected_facets=None, sortby=''):
     queryset = queryset.order_by(
         sortby, LoreSortingFields.BASE_SORTING_FIELD)
 
-    return queryset
+    # Do a parallel query using elasticsearch-dsl.
+    if query not in ("", None):
+        tokens = query
+    else:
+        tokens = None
+    terms = {}
+    for facet in selected_facets:
+        key, value = facet.split(":")
+
+        # Haystack queries for blanks by putting the key last, as the value,
+        # and setting the key to "_missing_." Fix  this for elasticsearch-dsl.
+        if key == '_missing_':
+            key, value = value, "empty"
+
+        if key.endswith("_exact"):
+            key = key[:-6]
+
+        # Look for facets
+        if key.isdigit():
+            key = get_vocab_name(key)
+            if value is not 'empty':
+                value = get_term_label(value)
+        terms[key] = value
```

Comment:
Maybe it's worth to put this code in a separated helper function that we can easily locate and remove if we ever decide to change the front-end and how the facets are passed to the back-end